### PR TITLE
eventhubs metadata propagation

### PR DESCRIPTION
**Pull request by tanvigour**
# Description

_Please explain the changes you've made_
Propagating metadata along with event.Data 

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #227

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Created/updated tests
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

**Comment by tanvigour** [comment]
@artursouza Please let me know your thoughts

File: bindings/azure/eventhubs/eventhubs.go [hunk 1]
```diff
@@ -198,10 +198,11 @@ func (a *AzureEventHubs) RegisterPartitionedEventProcessor(handler func(*binding
 		return err
 	}
 
-	callback := func(c context.Context, event *eventhub.Event) error {
+	callback := func(c context.Context, event *eventhub.Event, bind *bindings.InvokeResponse) error {
```
Comment (CodeMonkeyLeet) [inline bindings/azure/eventhubs/eventhubs.go:201]: Does this compile? The [eventhubs.Hub.Receive](https://pkg.go.dev/github.com/Azure/azure-event-hubs-go/v3#Hub.Receive) takes a callback of type `Handler func(ctx context.Context, event *Event) error`.

Also note that any changes to `ReadResponse` will also need to be reflected in `RegisterEventProcessor` as well.

Comment (daixiang0) [reply bindings/azure/eventhubs/eventhubs.go:201]: ping @tanvigour

**Comment by CodeMonkeyLeet** [comment]
@tanvigour @artursouza This can be closed as it's replaced by #1009 that has already been merged.

**Comment by tanvigour** [comment]
closing since this was already addressed by @CodeMonkeyLeet  in #1009

**Comment by artursouza** [comment]
Ack.

**Pull request closed without merging**